lib/rust/transformer/src/main.rs

```diff
@@ -663,6 +664,120 @@ impl LineLevelError {
     }
 }
 
+async fn write_publish_lines(
+    s3: aws_sdk_s3::Client,
+    sns: aws_sdk_sns::Client,
+    download_items: Vec<(DataBatcherOutputRecord, String)>,
+    mut stream: Pin<Box<dyn Stream<Item = Result<LineResult, LineError>> + Send>>,
+) -> Result<(
+    Vec<(String, String, String, usize, usize)>,
```

Comment:
Nit: Too much tuple, should use a struct

Reply:
agree its metrics stuff, will do sepretly,